### PR TITLE
Add Project.toml, drop dependency on Compat

### PR DESCRIPTION
**Pull request by giordano**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name = "BinDeps"
+uuid = "9e28174c-4ba2-5203-b857-d8d62c4213ee"
+version = "0.8.10"
+
+[deps]
+BinaryProvider = "b99e7846-7c00-51b0-8f62-c81ae34c0232"
```
Comment (oxinabox) [inline Project.toml]: thats new

Comment (oxinabox) [reply Project.toml]: It doesn't seem to be used?

Comment (giordano) [reply Project.toml]: I was getting an error without this:
```
ERROR: LoadError: LoadError: ArgumentError: Package BinaryProvider not found in current path:
- Run `import Pkg; Pkg.add("BinaryProvider")` to install the BinaryProvider package.

Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:876
 [2] include at ./boot.jl:328 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1094
 [4] include at ./Base.jl:31 [inlined]
 [5] eval_anon_module(::BinDeps.PackageContext, ::String) at /home/mose/.julia/dev/BinDeps/src/BinDeps.jl:557
 [6] debug_context(::String) at /home/mose/.julia/dev/BinDeps/src/debug.jl:52
 [7] debug(::Base.TTY, ::String) at /home/mose/.julia/dev/BinDeps/src/debug.jl:57
 [8] debug(::String) at /home/mose/.julia/dev/BinDeps/src/debug.jl:69
 [9] top-level scope at /home/mose/.julia/dev/BinDeps/test/runtests.jl:23
 [10] include at ./boot.jl:328 [inlined]
 [11] include_relative(::Module, ::String) at ./loading.jl:1094
 [12] include(::Module, ::String) at ./Base.jl:31
 [13] include(::String) at ./client.jl:431
 [14] top-level scope at none:5
in expression starting at /home/mose/.julia/packages/Cairo/ZbzIH/deps/build.jl:1
in expression starting at /home/mose/.julia/dev/BinDeps/test/runtests.jl:23
ERROR: Package BinDeps errored during testing
```
Maybe it's only used in tests?

Comment (giordano) [reply Project.toml]: It was needed only by (new) Cairo during test.  But we need old Cairo for testing.  Thanks for pointing this out

**Comment by mcabbott** [comment]
Bump? 

This would be nice to have, I think it’s causing tests to fail for me.

**Comment by musm** [comment]
agreed

**Comment by musm** [comment]
I propose to merge this today sans objections

**Comment by giordano** [comment]
Agreed.  CI for macOS will likely fail because installing Cairo with BinDeps has been being broken for months now, but this is unrelated with this PR, so I think we can go ahead anyway

**Comment by musm** [comment]
Cairo failures unrelated, thank you for doing this @giordano !

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name = "BinDeps"
+uuid = "9e28174c-4ba2-5203-b857-d8d62c4213ee"
+version = "0.8.10"
```
Comment (musm) [inline Project.toml]: @giordano I think we need to bump this to 0.9.0 since we drop julia  <1

**Pull request merged**